SocialCareCaseViewerApi.Tests/V1/UseCase/CaseStatus/CaseStatusExecutePostUseCaseTests.cs

```diff
@@ -0,0 +1,89 @@
+using FluentAssertions;
+using Moq;
+using NUnit.Framework;
+using SocialCareCaseViewerApi.Tests.V1.Helpers;
+using SocialCareCaseViewerApi.V1.Boundary.Requests;
+using SocialCareCaseViewerApi.V1.Exceptions;
+using SocialCareCaseViewerApi.V1.Gateways;
+using SocialCareCaseViewerApi.V1.Infrastructure;
+using SocialCareCaseViewerApi.V1.UseCase;
+using System.Collections.Generic;
+using System;
+
+namespace SocialCareCaseViewerApi.Tests.V1.UseCase.Relationships
+{
+    [TestFixture]
+    public class CaseStatusExecutePostUseCaseTests
+    {
+        private Mock<IDatabaseGateway> _mockDatabaseGateway;
+        private CaseStatusesUseCase _caseStatusesUseCase;
+        private CreateCaseStatusRequest _request;
+        private readonly CaseStatusType _typeInRequest = TestHelpers.CreateCaseStatusType();
+
+        [SetUp]
+        public void SetUp()
+        {
+            _mockDatabaseGateway = new Mock<IDatabaseGateway>();
+            _caseStatusesUseCase = new CaseStatusesUseCase(_mockDatabaseGateway.Object);
+
+            _mockDatabaseGateway.Setup(x => x.GetCaseStatusTypeWithFields(_typeInRequest.Name))
+                .Returns(_typeInRequest);
+
+            _request = CaseStatusHelper.CreateCaseStatusRequest(type: _typeInRequest.Name);
+
+            _mockDatabaseGateway.Setup(x => x.GetPersonByMosaicId(It.IsAny<int>()))
+                .Returns(TestHelpers.CreatePerson((int) _request.PersonId));
+
+            CaseStatus _caseStatus = CaseStatusHelper.CreateCaseStatus();
+
+            _mockDatabaseGateway.Setup(x => x.CreateCaseStatus(It.IsAny<CreateCaseStatusRequest>()))
+                .Returns(_caseStatus);
+
+            _mockDatabaseGateway.Setup(x => x.GetWorkerByEmail(It.IsAny<string>()))
+                .Returns(TestHelpers.CreateWorker(email: _request.CreatedBy));
+        }
+
+        [Test]
+        public void CallsDatabaseGatewayToCheckPersonExists()
```

Comment:
Should we test that all the functions are called correctly, considering nothing is returned this is the way to test our function works correctly?

Reply:
Added all the functions in this test